lua/telescope/actions/init.lua

```diff
@@ -1455,6 +1455,85 @@ end
 
 actions.nop = function(_) end
 
+actions.mouse_scroll_up = function(prompt_bufnr)
+  local picker = action_state.get_current_picker(prompt_bufnr)
+
+  local mouse_win = vim.fn.getmousepos().winid
+  if picker.results_win == mouse_win then
+    vim.defer_fn(function()
+      actions.move_selection_next(prompt_bufnr)
+    end, 0)
+    return ""
+  else
+    return "<ScrollWheelDown>"
+  end
+end
+
+actions.mouse_scroll_down = function(prompt_bufnr)
+  local picker = action_state.get_current_picker(prompt_bufnr)
+
+  local mouse_win = vim.fn.getmousepos().winid
+  if mouse_win == picker.results_win then
+    vim.defer_fn(function()
+      actions.move_selection_previous(prompt_bufnr)
+    end, 0)
+    return ""
+  else
+    return "<ScrollWheelUp>"
+  end
+end
+
+actions.mouse_scroll_right = function(prompt_bufnr)
+  local picker = action_state.get_current_picker(prompt_bufnr)
+
+  local mouse_win = vim.fn.getmousepos().winid
+  if mouse_win == picker.results_win then
+    return ""
+  else
+    return "<ScrollWheelLeft>"
+  end
+end
+
+actions.mouse_scroll_left = function(prompt_bufnr)
+  local picker = action_state.get_current_picker(prompt_bufnr)
+
+  local mouse_win = vim.fn.getmousepos().winid
+  if mouse_win == picker.results_win then
+    return ""
+  else
+    return "<ScrollWheelRight>"
+  end
+end
+
+actions.mouse_click = function(prompt_bufnr)
+  local picker = action_state.get_current_picker(prompt_bufnr)
+
+  local pos = vim.fn.getmousepos()
+  if pos.winid == picker.results_win then
+    vim.defer_fn(function()
+      picker:set_selection(picker:get_row(picker.max_results - pos.line + 1))
```

Comment:
this set selection part doesnt work if we are using `sorting_strategy = "ascending",`